src/scheduler/worker.rs

```diff
@@ -39,6 +42,24 @@ impl<C: Context> Worker<C> {
             stat: Default::default(),
             context: None,
             is_coordinator,
+            local_work_buffer: Vec::with_capacity(LOCALLY_CACHED_WORKS),
+        }
+    }
+
+    #[inline]
+    pub fn add_work(&mut self, bucket: WorkBucketId, work: impl Work<C>) {
+        self.local_work_buffer.push((bucket, box work));
```

Comment:
Do we allow adding work to any bucket (active and inactive buckets)?

Reply:
I feel we may only want to push works to local buffer for currently activated buckets. For works in other buckets, we should not put them to the local buffer, as they take up place in local buffer, and can't be done immediately.

Reply:
Fixed. Now it will cache work locally only when the corresponding bucket is open.